sn_interface/src/network_knowledge/mod.rs

```diff
@@ -198,17 +187,10 @@ impl NetworkKnowledge {
             SectionTree::new(genesis_signed_sap)?
         };
         let mut network_knowledge = Self::new(Prefix::default(), section_tree)?;
-
-        for peer in network_knowledge.signed_sap.elders() {
-            let node_state = NodeState::joined(*peer, None);
-            let sig = create_first_sig(&public_key_set, &secret_key_share, &node_state)?;
-            let _changed = network_knowledge
-                .section_peers
-                .update(section_signed_to_decision(SectionSigned {
-                    value: node_state,
-                    sig,
-                }));
-        }
+        let initial_members: BTreeSet<_> = iter::once(NodeState::joined(peer, None)).collect();
```

Comment:
I find the `from_iter(..)` pattern more clear:
```suggestion
        let initial_members = BTreeSet::from_iter([NodeState::joined(peer, None)]);
```
Either way works 👍

Reply:
`NodeState::joined(peer, None)` is not an iter, hence has to call some `into_iter` if want to using `from_iter` .
so still prefer to use `: BTreeSet<_>` in this case.